shopping-list-web/src/InputText/InputText.tsx

```diff
@@ -1,30 +1,70 @@
-import { Dispatch } from "react";
+import { Dispatch, useState, ChangeEvent, useEffect } from "react";
+import IListItem from "../ListItem/IListItem";
+import "./InputText.css"
+import { getDataResponseForFilteredProducts } from "../utilities/httpMethods/productMethods";
 
 export default function InputText(props: {
+  currentProduct: string;
+  currentAmount: string
   setCurrentProduct: Dispatch<React.SetStateAction<string>>;
   setCurrentAmount: Dispatch<React.SetStateAction<string>>;
   handleSubmit: () => Promise<void> | void;
 }) {
+
+  const [items, setItems] = useState<Map<string, IListItem[]>>(
+    new Map<string, IListItem[]>()
+  );
+
+  const handleProductInput = (e: ChangeEvent<HTMLInputElement>) => {
+    const current_value = e.target.value
+    props.setCurrentProduct(current_value)
+    
+  }
+
+  const handleAutoCompleteClick = (item: string) => {
+    props.setCurrentProduct(item)
+  }
+
+  useEffect(() => {
+    const timeOutId = setTimeout(() => getDataResponseForFilteredProducts(props.currentProduct).then((res) => setItems(res)), 500);
```

Comment:
I don't really know what's going on here but it's very functional and looks cool

Reply:
This means that it only sends a request off after 500 milliseconds after the user has finished typing